src/sourmash/search.py

```diff
@@ -157,12 +158,132 @@ def collect(self, score, match):
         self.threshold = max(self.threshold, score)
         return True
 
+def shorten_md5(md5):
+    return md5[:8]
+
+@dataclass
+class SearchResult:
+    """Base class for sourmash search results."""
+    query: SourmashSignature
+    match: SourmashSignature
+    search_scaled: int
+    similarity: float = None # would be great to know which similarity it is & get rid of this.
```

Comment:
so, there's some search result that returns None!? That seems bad.

Reply:
ok, I think I don't understand what's going on here with dataclasses, exactly 😆 . The errors are confusing!

I think what's happening is that there are places where `SearchResult` is getting initialized without any values (i.e. with values set to None) as part of it being the base class to the other classes.

But my guess is that you might want to have a common base class for all three `*Result` classes, and this base class should contain only the intersection of the fields used for all - which would be query and match, and not much else.

(You could still have GatherResult inherit from PrefetchResult because they do share quite a bit of info.)

Reply:
added `BaseResult` to solve this issue